### PR TITLE
Remove save part2

### LINKED ISSUE #2492: Deprecation of _reformat_hb in hbonds/hbond_analysis.py for v1.0
Currently going through and doing the required removals/deprecations for #1745 and I noticed that _reformat_hb was set for deprecation as part of #1339 

```
        .. deprecated:: 1.0
           This is a compatibility layer so that we can provide the same output
           in timeseries as before. However, for 1.0 we should make timeseries
           just return _timeseries, i.e., change the format of timeseries to
           the un-ambiguous representation provided in _timeseries.
```

It seems like some change is required here prior to v1.0.0, but it's unclear to me as to what.

### PR DESCRIPTION
**Pull request by IAlibay**
Partly addresses #1745 (completes `save()` removal with #2487 and #2486 )

Also addresses #2491, since the deprecation was unclear and it seems like there were no releases between the announced deprecation of `analysis/hbonds/hbond_analysis.py` and now, the deprecation was set as effective v1.0 and set for removal in v2.0 (this is just a placeholder date).

Fixes #2492 

Changes made in this Pull Request:
 - Set deprecation date for `analysis/hbonds/hbond_analysis.py`.
 - Removed `save_table` from :class:`HydrogenBondAnalysis`.
 - Removed `save_results` from :class:`HydrogenBondAutoCorrel`.
 - Removed `filename`, `store` and `save_result` from :class:`PSAnalysis`.
 - Deprecation notice for :meth:`HydrogenBondAnalysis._reformat_hb` has been removed, entire class will be removed in 2.0 (#2492 )

Things left to do:
 - [x] Confirm the removal date for `MDAnalysis/analysis/hbonds/hbond_analysis`.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by orbeckst** [comment]
Set removal date for hbonds to 2.0 — that’s the plan.

**Comment by orbeckst** [comment]
I didn’t check why CI fails. But ping me when you need a review.

**Comment by IAlibay** [comment]
@orbeckst this should be good for review now

File: package/MDAnalysis/analysis/hbonds/hbond_analysis.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 :Copyright: GNU Public License v3
 
 ..Warning:
-    This module will be deprecated in version 1.0.
+    This module will be deprecated in version 2.0.
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/hbonds/hbond_analysis.py]: Should be "will be removed in 2.0"

File: package/MDAnalysis/analysis/hbonds/hbond_analysis.py [hunk 2]
```diff
@@ -567,7 +572,8 @@ def __init__(self, universe, selection1='protein', selection2='all', selection1_
         super(HydrogenBondAnalysis, self).__init__(universe.trajectory, **kwargs)
 
         warnings.warn(
-            "This class will be deprecated in version 1.0."
+            "This class is deprecated as of MDANalysis version 1.0 and will "
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/hbonds/hbond_analysis.py]: MDAnalysis

File: package/MDAnalysis/analysis/psa.py [hunk 3]
```diff
@@ -1509,31 +1507,19 @@ def run(self, **kwargs):
...
-
-             .. deprecated:: 0.19.0
-                `store` will be removed together with :meth:`save_results` in 1.0.0.
-
-        filename : str
-             string, filename to save :attr:`PSAnalysis.D`
 
-             .. deprecated:: 0.19.0
-                `filename` will be removed together with :meth:`save_results` in 1.0.0.
+        .. deprecated:: 0.19.0
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/psa.py]: Remove this deprecation note – they are gone now, aren't they?

File: package/MDAnalysis/analysis/hbonds/hbond_analysis.py (hunk 1 above)
Comment (IAlibay) [reply package/MDAnalysis/analysis/hbonds/hbond_analysis.py]: Should be fixed in 4c7f557

File: package/MDAnalysis/analysis/hbonds/hbond_analysis.py (hunk 2 above)
Comment (IAlibay) [reply package/MDAnalysis/analysis/hbonds/hbond_analysis.py]: Fixed in 4c7f557

File: package/MDAnalysis/analysis/psa.py (hunk 3 above)
Comment (IAlibay) [reply package/MDAnalysis/analysis/psa.py]: Thanks for the clarification here, this got brought up in #2487 but we weren't sure what the appropriate approach was here. Fixed in 4c7f557 and I'll update #2487 accordingly.

**Comment by orbeckst** [comment]
Yay, less code!

**Pull request merged**